test/tests.js

```diff
@@ -392,114 +394,188 @@ function myDeepEqual(actual, expected, text) {
   return deepEqual(actual, expected, text);
 }
 
-function hex2arrayBuffer(data) {
+function hexToArrayBuffer(data) {
   var length = data.length / 2;
   var ret = new Uint8Array(length);
   for (var i = 0; i < length; ++i) {
     ret[i] = parseInt(data.substr(i * 2, 2), 16);
   }
   return ret.buffer;
 }
+function arrayBufferToHex(encoded) {
+  var hex = "";
+  var uint8Array = new Uint8Array(encoded);
+  for (var i = 0; i < uint8Array.length; ++i) {
+    var byte = uint8Array[i];
+    hex += (byte >> 4).toString(16);
+    hex += (byte & 0xf).toString(16);
+  }
+  return hex;
+}
 
-testcases.forEach(function(testcase) {
-  var name = testcase[0];
-  var data = testcase[1];
-  var expected = testcase[2];
-  var binaryDifference = testcase[3];
-
-  test(name, function() {
-    myDeepEqual(CBOR.decode(hex2arrayBuffer(data)), expected, "Decoding");
-    var encoded = CBOR.encode(expected);
-    myDeepEqual(CBOR.decode(encoded), expected, "Encoding (deepEqual)");
-    if (!binaryDifference) {
-      var hex = "";
-      var uint8Array = new Uint8Array(encoded);
-      for (var i = 0; i < uint8Array.length; ++i)
-        hex += (uint8Array[i] < 0x10 ? "0" : "") + uint8Array[i].toString(16);
-      equal(hex, data, "Encoding (byteMatch)");
-    }
-  });
-});
+for (var i = 0; i < testcases.length; ++i) {
+  try {
```

Comment:
using the closure of try { } catch ( x ) { } to ensure consistency of checks during async testing, equivalent to 'let'